exp/runtime/topologymutation/variables.go

```diff
@@ -17,39 +17,78 @@ limitations under the License.
 package topologymutation
 
 import (
+	"encoding/json"
 	"strconv"
+	"strings"
 
+	"github.com/pkg/errors"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 
 	patchvariables "sigs.k8s.io/cluster-api/internal/controllers/topology/cluster/patches/variables"
 )
 
-// TODO: Add func for validating received variables are of the expected types
-// TODO: Add func for converting complex variables into go structs and/or other basic types.
+// TODO: Add func for validating received variables are of the expected types.
 
 // GetVariable get the variable value.
-func GetVariable(templateVariables map[string]apiextensionsv1.JSON, variableName string) (*apiextensionsv1.JSON, bool, error) {
+func GetVariable(templateVariables map[string]apiextensionsv1.JSON, variableName string) (*apiextensionsv1.JSON, error) {
```

Comment:
I'm generally fine with dropping the bool. Seems cleaner and idiomatic to check the resulting error, but the function that folks can use to check if the error is a not found error is from an internal package. 

Should we export / alias that one?

Reply:
Make sense, let me export it

Reply:
Then should I use this new one to replace all other previous `patchvariables.IsNotFoundError(err)`

Reply:
I just replaced the tests for this variable.go so that user can reference as example

Reply:
I would say let's replace all usages (there are also a few more in variables_test.go)

Reply:
Cool